### PR TITLE
Create histograms

### PR DESCRIPTION
**Pull request by jethronap**
Pls have a look at this PR. Thank you.

### CODE REVIEW

File: src/test/java/visualizations/HistogramsTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package visualizations;
```
Comment (pockerman) [inline src/test/java/visualizations/HistogramsTest.java]: same as for PR: https://github.com/jethronap/JavaDataAnalysis/pull/12

**Review by pockerman — CHANGES_REQUESTED** [review]
pls consider my comments

**Comment by pockerman** [comment]
I think you branched off form branch: https://github.com/jethronap/JavaDataAnalysis/tree/create_visualization_classes and now all new branches need this merge...since the features are irrelevant I think it would have been better had you branched off from ```master```

File: src/test/java/visualizations/HistogramsTest.java (hunk 1 above)
Comment (jethronap) [reply src/test/java/visualizations/HistogramsTest.java]: ok!

**Comment by jethronap** [comment]
pls check out updated version.

File: src/main/java/dataloader/CsvDataLoader.java [hunk 2]
```diff
@@ -108,29 +110,27 @@ public static CategoricalSample buildCategoricalSample(Map<String, List<String>>
...
                             // add a new column
                             colNames.add(field);
                             dataSet.put(field, new ArrayList());
                         }
                     }
 
                     lineCounter++;
-                }
-                else{
+                } else {
```
Comment (pockerman) [inline src/main/java/dataloader/CsvDataLoader.java]: This is a big definitely NOOOOOO :) :) :) I absolutely detest this thing

File: src/main/java/datastructs/NumericSample.java [hunk 3]
```diff
@@ -2,210 +2,247 @@
...
+            this.data_.add(0.0);
+        }
+    }
+
+    /**
+     * Compute the sample statistics
+     */
+    protected void compute_sample_statistics() {
+
+        double[] arrayData = ArrayOperations.toArray(this.data_);
```
Comment (pockerman) [inline src/main/java/datastructs/NumericSample.java]: We expose an ```asArray``` function so we don't need the ```ArrayOperations``` here.

File: src/main/java/visualizations/ScatterChart.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * Plots a scatter chart given the chart title,
+     * data set in Table format,
+     * and two numeric columns
+     * created from a numeric sample
+     * of the data set.
+     */
+
+    public static void plotScatter(NumericSample x, NumericSample y, ScatterChartOptions options){
```
Comment (pockerman) [inline src/main/java/visualizations/ScatterChart.java]: The rest of the functions in this class pass first the options and then the data. Why did you decide that this one is different?

**Review by pockerman — CHANGES_REQUESTED** [review]
Pls consider my comments

File: src/main/java/dataloader/CsvDataLoader.java (hunk 2 above)
Comment (jethronap) [reply src/main/java/dataloader/CsvDataLoader.java]: I don't quite get what you are saying. please be more precise. and in any case I didn't do anything to this implementation

File: src/main/java/datastructs/NumericSample.java (hunk 3 above)
Comment (jethronap) [reply src/main/java/datastructs/NumericSample.java]: I think we have to merge create_visualization_classes into this branch.
maybe these are left from an earlier implementation.

File: src/main/java/visualizations/ScatterChart.java (hunk 4 above)
Comment (jethronap) [reply src/main/java/visualizations/ScatterChart.java]: reversed the order & updated documentation.

**Comment by jethronap** [comment]
pls see updated version and comments above and tell me what you think.

File: src/main/java/visualizations/ScatterChart.java [hunk 5]
```diff
@@ -23,14 +23,14 @@
     }
 
     /**
-     * Plots a scatter chart given the chart title,
+     * Plots a scatter chart given the chart options,
      * data set in Table format,
```
Comment (pockerman) [inline src/main/java/visualizations/ScatterChart.java]: I can't see ```Table``` argument. The fact that the function uses ```Table``` is irrelevant for the client code. Such things typically we should document them whenever there is some very good reason to do so for example thread safety, memory consumption compute time etc

**Review by pockerman — CHANGES_REQUESTED** [review]
Pls consider my comment

File: src/main/java/visualizations/ScatterChart.java (hunk 5 above)
Comment (jethronap) [reply src/main/java/visualizations/ScatterChart.java]: I see your point and corrected the documentation.

**Pull request merged**